Segfault on shutdown if etcd fails to start

If etcd fails to start, mke will segfault during shutdown:
```
...
INFO[2020-10-09 11:37:50] etcd on unsupported platform without ETCD_UNSUPPORTED_ARCH=arm64 set  component=etcd
WARN[2020-10-09 11:37:50] exit status 1                                 component=etcd
INFO[2020-10-09 11:37:50] respawning in 5 secs                          component=etcd
INFO[2020-10-09 11:37:50] Flag --insecure-port has been deprecated, This flag will be removed in a future version.  component=kube-apiserver
INFO[2020-10-09 11:37:50] I1009 11:37:50.622860  468151 server.go:625] external host was not specified, using 147.75.197.210  component=kube-apiserver
INFO[2020-10-09 11:37:50] I1009 11:37:50.624012  468151 server.go:163] Version: v1.19.0  component=kube-apiserver
^CINFO[2020-10-09 11:37:52] Shutting down mke server                     
INFO[2020-10-09 11:37:52] proxy reconciler done                         component=kubeproxy
INFO[2020-10-09 11:37:52] Shutting down pid 468084                      component=mke-control-api
INFO[2020-10-09 11:37:52] coredns reconciler done                       component=calico
INFO[2020-10-09 11:37:52] coredns reconciler done                       component=coredns
INFO[2020-10-09 11:37:52] metric server reconciler done                 component=metricServer
2020-10-09 11:37:52.254910 I | [INFO] lost leader lease
INFO[2020-10-09 11:37:52] Shutting down pid 467982                      component=kube-controller-manager
INFO[2020-10-09 11:37:52] lost leader lease                             component=applier-manager
panic: close of nil channel

goroutine 132 [running]:
github.com/Mirantis/mke/pkg/applier.(*Manager).watchLeaseEvents(0x400001a120, 0x400039d430)
        /home/ncopa/go/src/github.com/Mirantis/mke/pkg/applier/manager.go:100 +0xe4
created by github.com/Mirantis/mke/pkg/applier.(*Manager).Run
        /home/ncopa/go/src/github.com/Mirantis/mke/pkg/applier/manager.go:73 +0x1e0
```

I suppose it should be possible to reproduce by killing etcd, and then shutdown mke before etcd is restarted (5 seconds)

--- a/pkg/applier/manager.go
+++ b/pkg/applier/manager.go
@@ -20,8 +20,7 @@ type Manager struct {
 	client               kubernetes.Interface
 	applier              Applier
 	bundlePath           string
-	tickerDone           chan struct{}
-	watcherDone          chan struct{}
+	cancelWatcher        context.CancelFunc
 	cancelLeaderElection context.CancelFunc
 	log                  *logrus.Entry
 }
@@ -92,18 +91,19 @@ func (m *Manager) watchLeaseEvents(events *leaderelection.LeaseEvents) {
 		select {
 		case <-events.AcquiredLease:
 			log.Info("acquired leader lease")
-			m.tickerDone = make(chan struct{})
-			m.watcherDone = make(chan struct{})
-			go m.runFSWatcher()
+			ctx, cancel := context.WithCancel(context.Background())
+			m.cancelWatcher = cancel
+			go m.runFSWatcher(ctx)
 		case <-events.LostLease:
 			log.Info("lost leader lease")
-			close(m.tickerDone)
-			close(m.watcherDone)
+			if m.cancelWatcher != nil {
+				m.cancelWatcher()
+			}
 		}
 	}
 }
 
-func (m *Manager) runFSWatcher() {
+func (m *Manager) runFSWatcher(ctx context.Context) {
 	log := logrus.WithField("component", "applier-manager")
 	watcher, err := fsnotify.NewWatcher()
 	if err != nil {
@@ -135,7 +135,7 @@ func (m *Manager) runFSWatcher() {
 		select {
 		case err := <-watcher.Errors:
 			log.Warnf("watch error: %s", err.Error())
-		case <-m.watcherDone:
+		case <-ctx.Done():
 			log.Info("manifest watcher done")
 			return
 		}

